src/spa/cloudfront_config.ts

```diff
@@ -0,0 +1,72 @@
+import { Construct, CfnOutput } from '@aws-cdk/core';
+import { IBucket } from '@aws-cdk/aws-s3';
+import { IHostedZone, ARecord, RecordTarget } from '@aws-cdk/aws-route53';
+import { CloudFrontTarget } from '@aws-cdk/aws-route53-targets/lib';
+import {
+  CloudFrontWebDistribution,
+  SSLMethod,
+  SecurityPolicyProtocol
+} from '@aws-cdk/aws-cloudfront';
+
+export interface CloudFrontConfigProps {
+  subDomain?: string;
+  /**
+   * certificateArn: "arn:aws:acm:region:123456789012:certificate/12345678-1234-1234-1234-123456789012"
+   * This value can be entered manually from your own certficate stored with ACM
+   * or you can create Route53AcmSslStack.  
+   */
+  certificateArn: string;
+  /**
+   *  sourceBucket: IBucket
+   *  The S3 bucket where your live project is located.
+   */
+  sourceBucket: IBucket;
+  /**
+   *  hostedZone: IHostedZone
+   *  Route53 HostedZone created via Route53AcmSslStack
+   */
+  hostedZone: IHostedZone;
+}
+
+export default class CloudFrontConfig extends Construct {
```

Comment:
I would like these constructs to be as self sufficient as possible.  Right now, in order for you to use this construct, you have to provide:

- hosted zone (unavoidable)
- certificate ARN (understandable)
- source bucket

Let's create a source bucket if one isn't provided.  I almost think that we should do the same with the certificate, but I think for now a certificateArn is good.  As much as possible, I think a construct can construct something that is usable in one construct.=